### PR TITLE
R4ML new chapter

### PR DESCRIPTION
**Pull request by aloknsingh**
R4ML code after name changes and cleanup from previous repo

### CODE REVIEW

File: R4ML/R/ml.model.mlogit.R [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+  }
+)
+
+#' @title Predict method for Multinomial Logistic Regression models
+#' @name predict.r4ml.mlogit
+#' @description This method allows to score/test a Multinomial Logistic Regression model for a given r4ml.matrix. If the testing set is labeled,
+#' testing will be done and some statistics will be computed to measure the quality of the model. Otherwise, scoring will be performed
+#' and only the probabilities will be computed.
+#' @param object (r4ml.mlogit) :
+#'  A Multinomial Logistic Regression model built by Hydra R
```
Comment (bdwyer2) [inline R4ML/R/ml.model.mlogit.R]: Hydra R

File: R4ML/R/ml.model.pca.R [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+# overwrite the base model's post training function so that one can
+# post process the final outputs from the dml scripts
+setMethod(
+  "r4ml.model.postTraining",
+  signature = "r4ml.pca",
+  def =
+    function(model) {
+      outputs <- model@dmlOuts$sysml.execute
+      
+      #Converting hydra.matrix to data.frame for consumption
```
Comment (bdwyer2) [inline R4ML/R/ml.model.pca.R]: hydra

File: R4ML/R/ml.model.step.lm.R [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+##'\tab\code{call}          \tab (character) \tab String representation of this method's call,
+##'                                               including the parameters and values passed to it.\cr
+##'
+##' }
+#'
+#' @examples \dontrun{
+#'
+#' # Convert the iris data set into a R4ML frame.
+#' # Column #5 is a factor so we will ignore it for now
+#' hydra_iris <- as.r4ml.frame(iris[, -5])
```
Comment (bdwyer2) [inline R4ML/R/ml.model.step.lm.R]: hydra

File: R4ML/R/ml.model.step.lm.R [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+#' 
+#' # Convert the R4ML frame into a R4ML matrix
+#' hydra_iris <- as.r4ml.matrix(hydra_iris)
```
Comment (bdwyer2) [inline R4ML/R/ml.model.step.lm.R]: hydra

File: R4ML/R/ml.model.step.lm.R [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+#'
+#'
+#'step_lm <- r4ml.step.lm(Sepal_Length ~ ., data = hydra_iris)
```
Comment (bdwyer2) [inline R4ML/R/ml.model.step.lm.R]: hydra

File: R4ML/R/r4ml.matrix.R [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+      r4ml.err(logSource, "conversion from non numeric r4ml.frame is not supported yet")
+    }
+    result
+  }
+)
+
+setMethod("as.r4ml.matrix",
+  signature(object = "data.frame"),
+  function(object) {
+  hf <- as.r4ml.frame(object)
```
Comment (bdwyer2) [inline R4ML/R/r4ml.matrix.R]: should we change `hf` & `hm` to something else?

File: R4ML/tests/testthat/test-ml.model.lm.R [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  train <- aq_ozone_mat
+  aq_lm <- r4ml.lm(Ozone ~ . , data = train, method ="iterative")
+  aq_lm_ds <- r4ml.lm(Ozone ~ . , data = train, method ="direct-solve")
+  
+  show(aq_lm)
+  coef(aq_lm)
+  stats(aq_lm)
+})
+
+# test hydra linear model
```
Comment (bdwyer2) [inline R4ML/tests/testthat/test-ml.model.lm.R]: hydra

File: R4ML/tests/testthat/test-ml.model.lm.R [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+context("Testing r4ml.lm\n")
+
+# test hydra linear model
```
Comment (bdwyer2) [inline R4ML/tests/testthat/test-ml.model.lm.R]: hydra

**Comment by lresende** [comment]
I would do this without the last MERGE commit.

**Comment by lresende** [comment]
Basically, do this in a temporary fork, and then push --mirror...

**Comment by aloknsingh** [comment]
@lresende we have temporary fork at https://github.com/aloknsingh/r4ml. Merge is coming due to extra file README.md

@akchinSTC can you please remove README.md?

**Comment by lresende** [comment]
your fork still has the issue
Merge branch 'master' of ../Admin/r4ml/r4ml_stage

**Comment by bdwyer2** [comment]
All test cases pass on cluster

**Pull request closed without merging**